bin/mocha-current-file.js

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env node
+
+// Copyright IBM Corp. 2020. All Rights Reserved.
+// Node module: @loopback-next
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+/*
+================================================================================
+This is used in the launch.json to enable you to debug a test file written in
+typescript.  This function attempts to convert the passed typescript file to
+the best-guess output javascript file.
+
+It walks up the filesystem from the current file, stops at package.json, and
+looks in `dist`
+
+Ideally, we could somehow use the typescript compiler and tsconfig.json to get
+the explicit output file instead of trying to guess it, but there might be
+overhead.
+
+Ex:
+```jsonc
+{
+  "version": "0.2.0",
+  "configurations": [
+    {
+      "type": "node",
+      "request": "launch",
+      "name": "Debug Current Test File",
+      "program": "${workspaceRoot}/bin/mocha-current-file",
+      "runtimeArgs": ["-r", "${workspaceRoot}/packages/build/node_modules/source-map-support/register"],
+      "cwd": "${workspaceRoot}",
+      "autoAttachChildProcesses": true,
+      "args": [
+        "--config",
+        "${workspaceRoot}/packages/build/config/.mocharc.json",
+        "-t",
+        "0",
+        "${file}"
+      ],
+      "disableOptimisticBPs": true
+    }
+  ]
+}
+```
+================================================================================
+*/
+
+'use strict';
+const path = require('path');
+const fs = require('fs');
+
+function findDistFile(filename) {
+  const absolutePath = path.resolve(filename);
+  let currentDir = path.dirname(absolutePath);
+  let isPackageRoot = fs.existsSync(path.resolve(currentDir, 'package.json'));
+  while (!isPackageRoot) {
+    currentDir = path.join(currentDir, '..');
+    isPackageRoot = fs.existsSync(path.resolve(currentDir, 'package.json'));
+  }
+  const base = path.resolve(currentDir);
+  const relative = path.relative(currentDir, absolutePath);
+  const resultPath = relative.replace(/^src/, 'dist').replace(/\.ts$/, '.js');
+  return path.resolve(base, resultPath);
+}
```

Comment:
Nitpick: a more robust implementation can be:
```js
// Find root directory of the file system
const root = path.parse(process.cwd()).root;

let pkgFile = path.resolve(filename, '../package.json');
while (!fs.existsSync(pkgFile)) {
  if (path.dirname(pkgFile) === root) {
    // We reach the root of file system
    pkgFile = undefined;
    break;
  }
  // Try the parent directory level package.json
  pkgFile = path.resolve(pkgFile, '../../package.json');
}

if (!pkgFile) return filename; // Or throw an error
const projectRoot = path.dirname(pkgFile);
const relative = path.relative(projectRoot, path.resolve(filename));
const jsFilePath = relative.replace(/^src/, 'dist').replace(/\.ts$/, '.js');
return path.resolve(projectRoot, jsFilePath);
```

Reply:
I made a small change to the current function, and tested it by invoking this cli against both a test file inside of the `loopback-next` workspace, and also a `fake.test.js` file that has no `package.json` file in it's filesystem heirarchy